internal/internal_worker_base.go

```diff
@@ -260,7 +260,11 @@ func (bw *baseWorker) pollTask() {
 		if err != nil && enableVerboseLogging {
 			bw.logger.Debug("Failed to poll for task.", zap.Error(err))
 		}
-		if err != nil && isServiceTransientError(err) {
+		if err != nil {
+			if isNonRetriableError(err) {
+				bw.logger.Error("Worker stop polling task.", zap.Error(err))
+				return
```

Comment:
Is the intention here to stop all poller goroutines in the decision/activity worker? If so, is it possible to trigger a shutdown directly here (by calling `Stop()` maybe), instead of stopping the pollers one by one? Personally, I think that will be a little bit easier to understand. It will also stop all the poller goroutines instead of making them block on the `select` which will only be unblocked during worker shutdown.

Reply:
Make sense as discussed in team. Updated